nixos/modules/services/misc/ollama.nix

```diff
@@ -159,12 +169,64 @@ in
         OLLAMA_HOST = "${cfg.host}:${toString cfg.port}";
         HSA_OVERRIDE_GFX_VERSION = lib.mkIf (cfg.rocmOverrideGfx != null) cfg.rocmOverrideGfx;
       };
-      serviceConfig = {
+      serviceConfig = lib.optionalAttrs staticUser {
+        User = cfg.user;
+        Group = cfg.group;
+      } // {
+        DynamicUser = true;
```

Comment:
When `staticUser == true`, then `DynamicUser` will be enabled as well as `User` and `Group` being set. Are you sure that'll work right?

Reply:
Yes, that works:

> If a statically allocated user or group of the configured name already exists, it is used and no dynamic user/group is allocated.

https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#DynamicUser=